service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/CreateAndDeleteAzureControlledResourceFlightTest.java

```diff
@@ -497,6 +489,180 @@ public void createAzureStorageContainerFlightFailedBecauseLandingZoneDoesntExist
+    OffsetDateTime startTime = OffsetDateTime.now();
+    OffsetDateTime expiryTime = startTime.plusMinutes(15L);
+    var azureSasBundle =
+        azureStorageAccessService.createAzureStorageContainerSasToken(
+            workspaceUuid,
+            containerResourceId,
+            userRequest,
+            new SasTokenOptions(null, startTime, expiryTime, null, null));
+    assertNotNull(azureSasBundle);
+    assertNotNull(azureSasBundle.sasToken());
+    assertNotNull(azureSasBundle.sasUrl());
+
+    // clean up resources - delete storage container resource
+    submitControlledResourceDeletionFlight(
+        workspaceUuid,
+        userRequest,
+        containerResource,
+        azureTestUtils.getAzureCloudContext().getAzureResourceGroupId(),
+        containerResource.getStorageContainerName(),
+        null); // Don't sleep/verify deletion yet.
+
+    // clean up resources - delete storage account resource
+    submitControlledResourceDeletionFlight(
+        workspaceUuid,
+        userRequest,
+        accountResource,
+        azureTestUtils.getAzureCloudContext().getAzureResourceGroupId(),
+        accountResource.getStorageAccountName(),
+        azureTestUtils.getStorageManager().storageAccounts()::getByResourceGroup);
+
+    verifyStorageAccountContainerIsDeleted(accountResource, containerName);
+  }
+
+  @Test
+  public void generateAzureStorageContainerSasTokenBasedOnLandingZoneSharedStorageAccount()
+      throws InterruptedException {
+    // name of the shared storage account in landing zone
+    String storageAccountName = String.format("lzsharedsa%s", Instant.now().getEpochSecond());
+
+    Workspace workspace = azureTestUtils.createWorkspace(workspaceService);
+    UUID workspaceUuid = workspace.getWorkspaceId();
+    AuthenticatedUserRequest userRequest = userAccessUtils.defaultUserAuthRequest();
+    createCloudContext(workspaceUuid, userRequest);
+
+    // create quasi landing zone with a single resource - shared storage account
+    UUID landingZoneId = UUID.fromString(landingZoneTestUtils.getDefaultLandingZoneId());
+
+    TestLandingZoneManager testLandingZoneManager =
+        new TestLandingZoneManager(
+            azureCloudContextService,
+            landingZoneDao,
+            workspaceDao,
+            crlService,
+            azureConfig,
+            workspaceUuid);
+
+    testLandingZoneManager.createLandingZoneWithSharedStorageAccount(
+        landingZoneId, workspaceUuid, storageAccountName, "eastus");
+
+    TimeUnit.MINUTES.sleep(1);
+
+    // Submit a storage container creation flight and then verify the resource exists in the
+    // workspace.
+    final UUID containerResourceId = UUID.randomUUID();
+    final String storageContainerName = ControlledResourceFixtures.uniqueBucketName();
+    ControlledAzureStorageContainerResource containerResource =
+        ControlledAzureStorageContainerResource.builder()
+            .common(
+                ControlledResourceFields.builder()
```

Comment:
nit: building ControlledResourceFields object is repeated several times, may be taken to a private method or test fixture method.

Reply:
[ControlledResourceFixtures](https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/test/java/bio/terra/workspace/common/fixtures/ControlledResourceFixtures.java) might a good home for such code.